data-structures-for-storage-and-retrieval/skip-lists/skip_list_oc.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"math/rand"
+	"time"
+)
+
+const maxLevel = 10
+const p = 0.5
+
+type skipNode struct {
+	item    Item
+	forward []*skipNode
+}
+
+type skipListOC struct {
+	header *skipNode
+	tail   *skipNode
+	level  int
+}
+
+func newSkipListOC() *skipListOC {
+	header := &skipNode{
+		forward: make([]*skipNode, maxLevel+1),
+	}
+	tail := &skipNode{
+		Item{"zzzzzzz", "zzzzzzzz"},
```

Comment:
What if the user adds a key that's `{hello}`, since `{` comes after `z` in ASCII :)

Reply:
This is a great comment. I struggled with figuring out how to make the key in the tail greater than any possible user-entered key.

Since a `nil` string isn't permissible in Go, I added this as a hack.

Since 127 is the greatest ascii value, what are your thoughts on comparing it to: `string([]byte{127})`?